edward/data.py

```diff
@@ -30,57 +38,71 @@ def __init__(self, data=None, shuffled=True):
             # shuffle self.data
             raise NotImplementedError()
 
-        self.counter = 0
         if self.data is None:
-            self.N = None
+            pass
         elif isinstance(self.data, tf.Tensor):
             self.N = self.data.get_shape()[0].value
+            self.counter = 0
```

Comment:
i now see that counter wraps around the axis that is being subsampled... perhaps some jargon that involves the words `window` or `sliding` and perhaps the word `index` might be more intuitive?